lib/connection.js

```diff
@@ -196,6 +219,7 @@ Connection.prototype.createSocket = function () {
 		this.socket.on("data",    this.handleTransmissionError.bind(this, this.socket));
 		this.socket.on("drain",   this.socketDrained.bind(this, this.socket, true));
 		this.socket.once("close", this.socketClosed.bind(this, this.socket));
+                this.socket.deferredConnection = this.deferredConnection;
```

Comment:
Yes, I see that, but it was `this.deferredConnection` before, why did you change it? I don't understand how it's relevant to the min connections feature